docs/sources/setup-grafana/configure-docker.md

```diff
@@ -57,48 +91,217 @@ The following settings are hard-coded when launching the Grafana Docker containe
+
+# run the docker build command to build the image
+docker build
+  --build-arg "GRAFANA_VERSION=latest-ubuntu"
+  -t grafana-custom .
+
+# run the custom grafana container using docker run command
+docker run -d -p 3000:3000 --name=grafana grafana-custom
+```
+
+## Build a Grafana Docker image with pre-installed plugins
+
+To save time, you can customize a Grafana image by including plugins available on the [Grafana Plugin download page](/grafana/plugins). By doing so, you won't have to manually install the plugins each time, making the process more efficient.
+
+> **Note:** To specify the version of a plugin, you can use the `GF_INSTALL_PLUGINS` build argument and add the version number. The latest version is used if you don't specify a version number. For example, you can use `--build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"` to specify the versions of two plugins.
+
+Example:
+
+The following example shows how to build and run a custom Grafana image with pre-installed plugins.
+
+```bash
+# go to the custom directory
+cd packaging/docker/custom
+
+# running the build command
+# include the plugins you want e.g. clock planel etc
+docker build
+  --build-arg "GRAFANA_VERSION=latest"
+  --build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel,grafana-simple-json-datasource"
+  -t grafana-custom .
+
+# running the custom Grafana container using the docker run command
+docker run -d -p 3000:3000 --name=grafana grafana-custom
+```
+
+## Build a Grafana Docker image with pre-installed plugins from other sources
+
+You can create a Docker image containing a plugin that is exclusive to your organization, even if it is not accessible to the public. Simply use the `GF_INSTALL_PLUGINS` build argument to specify the plugin's URL and installation folder name, such as `GF_INSTALL_PLUGINS=;`.
+
+The following example demonstrates creating a customized Grafana Docker image that includes the clock panel and simple-json-datasource plugins. You can define these plugins in the build argument using the Grafana Plugin environment variable.
+
+```bash
+#go to the folder
+cd packaging/docker/custom
+
+#running the build command
+docker build
+--build-arg "GRAFANA_VERSION=latest"
+--build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel, grafana-simple-json-datasource"
+-t grafana-custom .
+
+# running the docker run command
+docker run -d -p 3000:3000 --name=grafana grafana-custom
+```
+
+## Build Grafana with the Image Renderer plugin pre-installed
+
+> **Note:** This feature is experimental.
+
+Currently, the Grafana Image Renderer plugin is not functional (as stated in [GitHub Issue#301](https://github.com/grafana/grafana-image-renderer/issues/301)) when installed in a Grafana Docker image. However, you can create a customized Docker image utilizing the `GF_INSTALL_IMAGE_RENDERER_PLUGIN` build argument as a solution. This will install the necessary dependencies for the Grafana Image Renderer plugin to run.
```

Comment:
```suggestion
Currently, the Grafana Image Renderer plugin requires dependencies which are not available in the Grafana Docker image (see [GitHub Issue#301](https://github.com/grafana/grafana-image-renderer/issues/301) for more details). However, you can create a customized Docker image utilizing the `GF_INSTALL_IMAGE_RENDERER_PLUGIN` build argument as a solution. This will install the necessary dependencies for the Grafana Image Renderer plugin to run.
```